base/inference.jl

```diff
@@ -1297,7 +1297,7 @@ function ⊑(a::ANY, b::ANY)
     end
 end
 
-widenconst(c::Const) = typeof(c.val)
+widenconst(c::Const) = type_typeof(c.val)
```

Comment:
K. I think I was probably confused by the printing and thought it could be.

```jl
julia> Core.Inference.print(Core.Inference.Const(sin))
Core.Inference.Const(val=Base.#sin())
```